src/idfxt/JSONDataInterface.h

```diff
@@ -0,0 +1,102 @@
+// copyright 2014, Shannon Mackey <mackey@BUILDlab.net>
+#ifndef JSONDATAINTERFACE_H
+#define JSONDATAINTERFACE_H
+
+//for JSONDataObject and JSONDataInterface
+#include <string>
+#include <list>
+#include <map>
+#include <memory>
+
+//for old IDF text import only
+#include <algorithm>
+#include <cctype>
+#include <cstring>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <vector>
+#include <chrono>
+#include <random>
+
+const std::string ACCEPTED_VERSION = "8.2";
+//end old IDF text import only
+
+extern "C"
+{
+#include "../../third_party/cJSON/cJSON.h"
+}
+
+namespace idfx {
+
+
+class JSONDataObject
+{
+public:
+    JSONDataObject(const std::string &json_content);
+    ~JSONDataObject();
+
+    std::map<std::string, std::string> getProperties();
+    std::string getPropertyValue(std::string);
```

Comment:
@refaqtor  I would add some accessors to get properties as doubles or ints in this interface, that would consolidate all the string->double and string->int conversion code in this class.